cmd/bnsd/x/blueaccount/handler.go

```diff
@@ -0,0 +1,519 @@
+package blueaccount
+
+import (
+	"regexp"
+
+	weave "github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/gconf"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+func RegisterQuery(qr weave.QueryRouter) {
+	NewDomainBucket().Register("domain", qr)
+	NewAccountBucket().Register("account", qr)
+}
+
+func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
+	r = migration.SchemaMigratingRegistry("blueaccount", r)
+
+	domains := NewDomainBucket()
+	accounts := NewAccountBucket()
+
+	r.Handle(&RegisterDomainMsg{}, &registerDomaiHandler{
+		domains:  domains,
+		accounts: accounts,
+		auth:     auth,
+	})
+	r.Handle(&TransferDomainMsg{}, &transferDomaiHandler{
+		domains: domains,
+		auth:    auth,
+	})
+	r.Handle(&RenewDomainMsg{}, &renewDomaiHandler{
+		domains: domains,
+		auth:    auth,
+	})
+	r.Handle(&DeleteDomainMsg{}, &deleteDomaiHandler{
+		domains: domains,
+		auth:    auth,
+	})
+	r.Handle(&RegisterAccountMsg{}, &registerAccounHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+	r.Handle(&TransferAccountMsg{}, &transferAccounHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+	r.Handle(&ReplaceAccountTargetsMsg{}, &replaceAccountTargetHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+	r.Handle(&DeleteAccountMsg{}, &deleteAccounHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+
+	r.Handle(&UpdateConfigurationMsg{}, gconf.NewUpdateConfigurationHandler(
+		"blueaccount", &Configuration{}, auth))
+}
+
+type registerDomaiHandler struct {
+	auth     x.Authenticator
+	domains  orm.ModelBucket
+	accounts orm.ModelBucket
+}
+
+func (h *registerDomaiHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
+	if _, err := h.validate(ctx, db, tx); err != nil {
+		return nil, err
+	}
+	return &weave.CheckResult{GasAllocated: 0}, nil
```

Comment:
I think domain registration must cost a fee. https://github.com/iov-one/weave/blob/master/cmd/bnsd/x/username/handlers.go#L36

Reply:
I don't understand what those feels mean anymore. We have [`x/msgfee`](https://github.com/iov-one/weave/tree/master/x/msgfee) to handle both anti spam and product defined fees. Do you know why do we need additional hardcoded fee?

Reply:
The outcome of the discussion is very productive and this should be documented as well.